test/e2e/chaos_test.go

```diff
@@ -71,22 +74,14 @@ func testKillRandomMembers(parentCtx context.Context, client crclient.Client, cl
 			},
 			Data: map[string]string{"value": string(value)},
 		}
-		var previousCreationError string
-		err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 30*time.Minute, true, func(ctx context.Context) (bool, error) {
-			if err := guestClient.Create(ctx, cm); err != nil {
-				if err.Error() != previousCreationError {
-					t.Logf("failed to create marker ConfigMap: %v", err)
-					previousCreationError = err.Error()
-				}
-				return false, nil
-			}
-			return true, nil
-		})
-		g.Expect(err).NotTo(HaveOccurred(), "failed to create marker configmap")
+		e2eutil.EventuallyObject(t, ctx, "create marker", func(ctx context.Context) (*corev1.ConfigMap, error) {
+			err := guestClient.Create(ctx, cm)
+			return cm, err
+		}, e2eutil.NoOpPredicate[*corev1.ConfigMap]())
```

Comment:
I'm not thrilled at how users need to explicitly say they have no predicate, open to ideas here. We already have options pattern - perhaps we have flavors of this func that make the distinction explicit in their name?